packages/react-ui-core/src/Modal/ClosableModal.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import autobind from 'autobind-decorator'
+
+import { Modal } from '../Modal'
+
+@themed(/^ClosableModal/)
+export default class ClosableModal extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    theme: PropTypes.object,
+    isOpen: PropTypes.bool,
+    onClose: PropTypes.func,
+    children: PropTypes.any,
+  }
+
+  static defaultProps = {
+    theme: {},
+    isOpen: false,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      isOpen: props.isOpen,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      this.setState({ isOpen: nextProps.isOpen })
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextProps.isOpen !== this.props.isOpen ||
+      nextProps.children !== this.props.children ||
+      nextState.isOpen !== this.state.isOpen
+  }
+
+  @autobind
+  closeModal() {
```

Comment:
My preference is to name all handlers starting with "handle", so `handleClose` or `handleCloseModal` in this case.

Reply:
Okay.